src/state.ts

```diff
@@ -11,37 +11,56 @@ import * as vscode from 'vscode';
  * invalid states.
  */
 export class StateManager {
-  constructor(readonly extensionContext: vscode.ExtensionContext) {}
+  private readonly latestPrefix: string = 'latest.';
+
+  constructor(readonly extensionContext: vscode.ExtensionContext, readonly folder: vscode.WorkspaceFolder) {}
+
+  private _get<T>(key: string): T | undefined {
+    const actualKey = this.folder.uri.fsPath + key;
+    let value = this.extensionContext.workspaceState.get<T>(actualKey);
+    if (value) {
+      this.extensionContext.globalState.update(this.latestPrefix + actualKey, value);
```

Comment:
globalState should only look at "key" or else it will be unique for all folders which is not what we want.

Reply:
Got it, had some misunderstanding.

Reply:
Wait but if we make this key, then the kit selection menu will never pop up again automatically when the project is firstly opened...

Reply:
☹️ I guess we'll have to revert it back to use globalState.  Sorry for dragging you through that.

Reply:
Isn't this way better though as it allows the user to use different settings in different workspace for the same folder?

Reply:
Discussed offline. Go back to the original way.